core/src/main/java/io/github/fourlastor/game/coordinates/Hex.java

```diff
@@ -65,4 +64,67 @@ public void set(Hex other) {
     public boolean sameAxisAs(Hex hex) {
         return hex.cube.x == cube.x || hex.cube.y == cube.y || hex.cube.z == cube.z;
     }
+
+    public Hex offset(Direction direction, int amount) {
+        return new Hex(cube.cpy().add(direction.cube.cpy().scl(amount)));
+    }
+
+    public enum Direction {
+        S(0, -1, 1),
+        SE(1, -1, 0),
+        NE(1, 0, -1),
+        N(0, +1, -1),
+        NW(-1, 1, 0),
+        SW(-1, 0, 1),
+        NONE(0, 0, 0),
+        ;
+
+        private final GridPoint3 cube;
+
+        Direction(int x, int y, int z) {
+            cube = new GridPoint3(x, y, z);
+        }
+
+        public static Direction byValue(GridPoint3 cube) {
```

Comment:
I haven't found usages of this in #33 - I think the by name and angle make sense, but I'd remove this and the x,y,z variant

Reply:
opposite() is using it (for convenience)

Reply:
Ah yep, I missed it, that makes sense :+1: